src/main/java/tech/toparvion/analog/util/config/ChoicesCustomConfigurationLoader.java

```diff
@@ -0,0 +1,78 @@
+package tech.toparvion.analog.util.config;
+
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.boot.env.PropertySourceLoader;
+import org.springframework.boot.env.YamlPropertySourceLoader;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+/**
+ * Environment post processor capable of loading choices list property from custom file.<br/>
```

Comment:
I think "property" word can be omitted in this context:
```suggestion
 * Environment post processor capable of loading choices list from a custom file.<br/>
```